scripts/polyrith_sage_helper.py

```diff
@@ -32,3 +33,6 @@ def monomial_to_string(etuple: ETuple, coeff: QQ) -> str:
 
 def polynomial_to_string(p) -> str:
     return sum_to_string(monomial_to_string(pows, coeff) for pows, coeff in p.dict().items())
+
+def naive_power_search(p, I) -> int:
+    return next(i for i in count(2) if p^i in I)
```

Comment:
For what it's worth, I think the `in` computation happens here:

https://github.com/sagemath/sage/blob/5fb2a6ea44400e469caee82748cf863ca0c5f724/src/sage/rings/polynomial/multi_polynomial_ideal.py#L3447